src/testdrive/src/action/s3.rs

```diff
@@ -93,3 +100,172 @@ impl Action for PutObjectAction {
+                    })
+                    .await
+                    .map_err(|e| {
+                        format!(
+                            "when trying to get sqs queue url for already-existing queue: {}",
+                            e
+                        )
+                    })?;
+                resp.queue_url
+                    .expect("successfully getting the url gets the url")
+            }
+            Err(e) => return Err(e.to_string()),
+        };
+
+        let queue_arn = state
+            .sqs_client
+            .get_queue_attributes(GetQueueAttributesRequest {
+                attribute_names: Some(vec!["QueueArn".to_string()]),
+                queue_url: queue_url.clone(),
+            })
+            .await
+            .map_err(|e| format!("getting queue {} attributes: {}", self.queue, e))?
+            .attributes
+            .expect("the result should not be empty")
+            .remove("QueueArn")
+            .unwrap();
+
+        // Configure the queue to allow the S3 bucket to write to this queue
+        let mut attributes = HashMap::new();
+        attributes.insert(
+            "Policy".to_string(),
+            allow_s3_policy(&queue_arn, &self.bucket_prefix, &state.aws_account),
+        );
+        state
+            .sqs_client
+            .set_queue_attributes(SetQueueAttributesRequest {
+                queue_url: queue_url.clone(),
+                attributes,
+            })
+            .await
+            .map_err(|e| format!("setting aws queue attributes: {}", e))?;
+
+        state.sqs_queues_created.insert(queue_url.clone());
+
+        // Configure the s3 bucket to write to the queue, without overwriting any existing configs
+        let mut config = state
+            .s3_client
+            .get_bucket_notification_configuration(GetBucketNotificationConfigurationRequest {
+                bucket: self.bucket.clone(),
+                ..Default::default()
+            })
+            .await
+            .map_err(|e| format!("getting bucket notification_configuration: {}", e))?;
+
+        {
+            let queue_configs = config.queue_configurations.get_or_insert_with(Vec::new);
+
+            queue_configs.push(QueueConfiguration {
+                events: self.events.clone(),
+                queue_arn,
+                ..Default::default()
+            });
+        }
+
+        let result = state
+            .s3_client
+            .put_bucket_notification_configuration(PutBucketNotificationConfigurationRequest {
+                bucket: self.bucket.clone(),
+                notification_configuration: config,
+                ..Default::default()
+            })
+            .await
+            .map_err(|e| {
+                format!(
+                    "Putting s3 bucket configuration notification {} \n{:?}",
+                    e, e
+                )
+            })?;
+        eprintln!("created bucket notification configuration: {:?}", result);
```

Comment:
(optional: testdrive convention is to print to stdout for stuff like this.)

Reply:
this is debug, I meant to remove it.